### PR TITLE
Add whats new page in toolkits

### PR DESCRIPTION
**Pull request by Flanker32**
Add support for whats new pages, it will shown when user install toolkit or invoke Tools-> Azure-> Whats New

image

> This PR only for contains logic for show whats new, the document will be commit in other pr before release.

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle [hunk 1]
```diff
@@ -47,7 +47,7 @@ intellij {
     pluginName = 'azure-toolkit-for-intellij'
     version = intellij_version
     updateSinceUntilBuild = Boolean.valueOf(updateVersionRange)
-    plugins = ['java', 'maven', dep_plugins, "properties"]
+    plugins = ['java', 'maven', dep_plugins, "properties",'markdown']
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle:50]: add a space before markdown

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum WhatsNewHelper {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java:48]: What's a good technical trick for singleton! No offense, but may I say I don't like the smell? Enum type can't be inheritted and hard to mock for unit testing... 😛

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            throws IOException {
+        final LightVirtualFile virtualFile = new LightVirtualFile(AZURE_TOOLKIT_FOR_JAVA);
+        virtualFile.setLanguage(Language.findLanguageByID("Markdown"));
+        virtualFile.setBinaryContent(content.getBytes());
+        virtualFile.putUserData(WHAT_S_NEW_ID, WHAT_S_NEW_CONSTANT);
+        final FileEditor[] fileEditors = fileEditorManager.openFile(virtualFile, true, true);
+        for (FileEditor fileEditor : fileEditors) {
+            if (fileEditor instanceof MarkdownSplitEditor) {
+                // Switch to markdown preview panel
+                ((MarkdownSplitEditor) fileEditor).triggerLayoutChange(SplitFileEditor.SplitEditorLayout.SECOND,
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java:86]: Shall we check user data for what's new editor here? I'm wonder if other Markdaow editor opened will be impacted or not.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    INSTANCE;
+
+    private static final String AZURE_TOOLKIT_FOR_JAVA = "Azure Toolkit for Java";
+    private static final String AZURE_TOOLKIT_WHATS_NEW = "AzureToolkit.WhatsNew";
+    private static final String VERSION_PATTERN = "<!-- Version: (.*) -->";
+    private static final String WHAT_S_NEW_CONSTANT = "WHAT_S_NEW";
+    private static final String WHAT_S_NEW_CONTENT_PATH = "/whatsnew/whatsnew.md";
+    private static final Key<String> WHAT_S_NEW_ID = new Key<>(WHAT_S_NEW_CONSTANT);
+
+    public synchronized void showWhatsNew(boolean force, Project project) throws IOException {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java:58]: I notice there are `NotNull` and `Nullable` annotations in the code above, how about also apply to this file? Btw, in my practise, we can implicit the type is `NotNull` which means only have to annotate `Nullable`.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java (hunk 3 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java]: Changed to use our own file editor to show markdown notes, so should not break users experience.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java (hunk 4 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java]: Added the `NotNull` mark to public method

**Pull request closed without merging**